src/components/common/SpeciesSelector/index.tsx

```diff
@@ -44,11 +45,11 @@ export default function SpeciesSelector<T extends { speciesId?: number } | undef
         setSpeciesList(response.sort((a, b) => a.scientificName.localeCompare(b.scientificName)));
       }
     },
-    [selectedOrganization.id]
+    [selectedOrganization, id]
```

Comment:
At first I thought this was a malformed edit that replaced a period with a comma, but turns out that `id` is a property of the component and was missing as a dependency here.